test/scripts/test-ovnkube-trace.sh

```diff
@@ -0,0 +1,472 @@
+}
+
+parse_args "$@"
+
+if [ ! -f ${OVNKUBE_TRACE} ]; then
+  echo "ovnkube trace binary not found"
+  exit 1
+fi
+
+if [ ${SETUP} == true ]; then
+  cleanup
+  spawn_deployment
+  spawn_hostnetwork_deployment
+  spawn_service
+  spawn_egressip
+fi
+
+src_pods=()
+dst_pods=()
+src_pods_hostnetwork=()
+dst_pods_hostnetwork=()
+egressip_src_pods=()
+dst_svc=( "-dst-namespace ovn-kubetrace-service-test -service service-test" )
+
+for namespace in ovn-kubetrace-pod-test; do 
+  pods=$(${KUBECTL} get pods -o name -n $namespace | sed 's#pod/##')
+  for pod in ${pods}; do
+    src_pods+=( "-src-namespace ${namespace} -src ${pod}" )
+    dst_pods+=( "-dst-namespace ${namespace} -dst ${pod}" )
+  done
+done
+
+for namespace in egressip-kubetrace-pod-test; do 
+  pods=$(${KUBECTL} get pods -o name -n $namespace | sed 's#pod/##')
+  for pod in ${pods}; do
+    egressip_src_pods+=( "-src-namespace ${namespace} -src ${pod}" )
+  done
+done
+
+for namespace in ovn-kubetrace-pod-test-hostnetwork; do 
+  pods=$(${KUBECTL} get pods -o name -n $namespace | sed 's#pod/##')
+  for pod in ${pods}; do
+    src_pods_hostnetwork+=( "-src-namespace ${namespace} -src ${pod}" )
+    dst_pods_hostnetwork+=( "-dst-namespace ${namespace} -dst ${pod}" )
+  done
+done
+
+if [ $RUN_DEST_IP_TESTS == true ]; then
+  echo "Run ovnkube-trace from all pod-test pods to 8.8.8.8"
+  for ((i = 0; i < ${#src_pods[@]}; i++)); do
+    for protocol in tcp udp; do
+      ovnkube_trace ${src_pods[$i]} -dst-ip 8.8.8.8 -${protocol}
+    done
+  done
+
+  echo "Run ovnkube-trace from all egressip-pod-test pods to 8.8.8.8"
+  for ((i = 0; i < ${#egressip_src_pods[@]}; i++)); do
+    for protocol in tcp udp; do
+      ovnkube_trace ${egressip_src_pods[$i]} -dst-ip 8.8.8.8 -${protocol}
+    done
+  done
+  
+  echo "Run ovnkube-trace from a pod-test-hostnetwork pod to 8.8.8.8 - this should fail"
+  if ovnkube_trace ${src_pods_hostnetwork[0]} -dst-ip 8.8.8.8 -${protocol}; then
+    echo "The result of this test should be a failure condition"
+    exit 1
+  fi
+fi
+
+echo "Run ovnkube-trace from all pod-test pods to all other pod-test pods"
+for ((i = 0; i < ${#src_pods[@]}; i++)); do
+  for ((j = 0; j < ${#dst_pods[@]}; j++)); do
+    for protocol in tcp udp; do
+      ovnkube_trace ${src_pods[$i]} ${dst_pods[$j]} -${protocol}
+    done
+  done
+done
+
+echo "Run ovnkube-trace from all pod-test pods to all pod-test-hostnetwork pods"
+for ((i = 0; i < ${#src_pods[@]}; i++)); do
+  for ((j = 0; j < ${#dst_pods_hostnetwork[@]}; j++)); do
+    for protocol in tcp udp; do
+      ovnkube_trace ${src_pods[$i]} ${dst_pods_hostnetwork[$j]} -${protocol}
+    done
+  done
+done
+
+echo "Run ovnkube-trace from all pod-test pods to the service-test service"
+for ((i = 0; i < ${#src_pods[@]}; i++)); do
+  for ((j = 0; j < ${#dst_svc[@]}; j++)); do
+    # udp with services is not a valid option
+    for protocol in tcp; do
+      ovnkube_trace ${src_pods[$i]} ${dst_svc[$j]} -${protocol}
+    done
+  done
+done
+
```

Comment:
Would it be useful to make sure udp to service does not work?
```
echo "Run ovnkube-trace from all pod-test pods to the service-test service on udp - this should fail"
if ovnkube_trace ${src_pods[0]} ${dst_svc[0]} -udp; then
  echo "The result of this test should be a failure condition"
  exit 1
fi
```

Reply:
done